deploy/cdk/marble-web-kiosk-export-pipeline/lib/marble-web-kiosk-export-pipeline-stack.ts

```diff
@@ -0,0 +1,259 @@
+            defaultInput: 'ndlib',
+          })
+        const repoBranch = this.node.tryGetContext('repoBranch') ||
+          readlineSync.question('Which branch do you want to build from? (default: master) ', {
+            defaultInput: 'master',
+          })
+
+        // S3 BUCKET FOR STORING ARTIFACTS
+        const artifactBucket = new Bucket(this, 'ArtifactBucket', {
+          removalPolicy: RemovalPolicy.DESTROY,
+        })
+        artifactBucket.addToResourcePolicy(new PolicyStatement({
+          principals: [new AnyPrincipal()],
+          effect: Effect.DENY,
+          actions: ['s3:*'],
+          conditions: {
+            'Bool': { 'aws:SecureTransport': false }
+          },
+          resources: [artifactBucket.bucketArn + '/*']
+        }))
+
+        // SNS TOPICS
+        const approvalTopic = new sns.Topic(this, 'PipelineApprovalTopic', {
+          displayName: 'PipelineApprovalTopic',
+        })
+
+        // IAM ROLES
+        const codepipeline_role = new Role(this, 'CodePipelineRole', {
+          assumedBy: new ServicePrincipal('codepipeline.amazonaws.com')
+        })
+
+        const codebuild_role = new Role(this, 'CodeBuildTrustRole', {
+          assumedBy: new ServicePrincipal('codebuild.amazonaws.com')})
+          // Allow checking what policies are attached to this role and any roles the application stack creates
+          // When cdk makes roles, it removes all punctuation. The application stack should prefix the LogicalId
+          // of any construct that needs roles with the stack name so that it matches the wildcard here.
+          const rolePattern = localStackName.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, "") + '*'
+          codebuild_role.addToPolicy(new PolicyStatement({
+            resources: [
+              codebuild_role.roleArn,
+              Fn.sub('arn:aws:iam::${AWS::AccountId}:role/' + rolePattern),
+            ],
+            actions: ['iam:GetRolePolicy'],
+          }))
+
+        // Allow access to Parameter Store
+        codebuild_role.addToPolicy(new PolicyStatement({
+            resources:[Fn.sub('arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:*')],
+            actions: ["ssm:GetParameterHistory",
+                "ssm:GetParametersByPath",
+                "ssm:GetParameters",
+                "ssm:GetParameter",
+            ],
+        }))
+        // Allow testing to send email
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:ses:${AWS::Region}:${AWS::AccountId}:identity/nd.edu') ],
+          actions: ['ses:SendEmail'],
+        }))
+
+        // Allow DescribeRule
+        codebuild_role.addToPolicy(new PolicyStatement({
+          resources: [ Fn.sub('arn:aws:events:${AWS::Region}:${AWS::AccountId}:rule/' + localStackName + '-*') ],
+          actions: ['events:DescribeRule', 'events:PutRule', 'events:PutTargets'],
+        }))
+
+        // Allow logging
+        codebuild_role.addToPolicy(new PolicyStatement({
```

Comment:
Not a request for change here, just a note that will make things easier in the future: A lot of these permissions are automagically handled by cdk for you if you allow it to initially create the role when you instantiate a project. You can then just attach additional custom policies that you need by `deployToTestProject.addToRolePolicy(new PolicyStatement({...})`